specs/queue.spec.js

```diff
@@ -0,0 +1,284 @@
+                    queue1: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19],
+                    queue2: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19],
+                }, {
+                    queue1: [1, 99],
+                    queue2: [100, 999],
+                },
+            ].forEach(data => {
+                let queue1 = new Queue(100);
+                queue1.queue = data.queue1;
+                let queue2 = new Queue(50);
+                queue2 = data.queue2;
+                let queue3 = new Queue(100);
+                queue3 = [...data.queue2, ...data.queue1];
+
+                expect(queue1.merge(queue1, true)).to.be.equal(queue3);
+            });
+        });
+
+        it('should reduce data to the size of first queue', function () {
+            [
+                {
+                    queue1: [1, 2, 3, 4, 5],
+                    queue2: [6, 7, 8, 9, 10],
+                }, {
+                    queue1: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31],
+                    queue2: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31],
+                }, {
+                    queue1: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19],
+                    queue2: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19],
+                }, {
+                    queue1: [1, 99],
+                    queue2: [100, 999],
+                },
+            ].forEach(data => {
+                const queue1MaxSize = data.queue1.length + Math.floor(data.queue2.length/2);
+                let queue1 = new Queue(queue1MaxSize);
+                queue1.queue = data.queue1;
+                let queue2 = new Queue(50);
+                queue2 = data.queue2;
+                let queue3 = new Queue(100);
+                queue3 = [...data.queue1, ...data.queue2].slice(0, queue1MaxSize + 1);
+
+                expect(queue1.merge(queue1, false)).to.be.equal(queue3);
+            });
+        });
+
+        it('should duplicate values, when merge queue with itself', function () {
+            [
+                {
+                    values: [1, 2, 3, 4, 5],
+                }, {
+                    values: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31],
+                }, {
+                    values: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19],
+                }, {
+                    values: [1, 99],
+                },
+            ].forEach(data => {
+                let queue1 = new Queue(100);
+                queue1.queue = data.values;
+                let queue2 = new Queue(100);
+                queue2.queue = [...data.values, ...data.values];
+
+                expect(queue1.merge(queue1, false)).to.be.equal(queue2);
+            });
+        });
+
+        it('should get empty queue, when two queues are empty', function () {
+            let queue1 = new Queue(100);
+            let queue2 = new Queue(100);
+            let queue3 = new Queue(100);
+
+            expect(queue1.merge(queue1, false)).to.be.equal(queue3);
+        });
+    });
+});
```

Comment:
most of the tests are testing the same logic several times that is not needed, while some functionality is not covered.